android/src/main/java/com/swmansion/rnscreens/ScreenFragment.java

```diff
@@ -66,6 +64,63 @@ public Screen getScreen() {
     return mScreenView;
   }
 
+  public void onContainerUpdate() {
+    ScreenStackHeaderConfig config = findHeaderConfig();
+    boolean configInChild = hasChildScreenWithConfig(getScreen());
+    if (!configInChild && config != null && config.getScreenFragment().getActivity() != null) {
+      // if there is no child with config, we look for a parent with config to set the orientation
+      config.getScreenFragment().getActivity().setRequestedOrientation(config.getScreenOrientation());
+    }
+  }
+
+  private ScreenStackHeaderConfig findHeaderConfig() {
+    ViewParent parent = getScreen().getContainer();
+    while (parent != null) {
+      if (parent instanceof Screen) {
+        ScreenStackHeaderConfig headerConfig = ((Screen) parent).getHeaderConfig();
+        if (headerConfig != null) {
+          return headerConfig;
+        }
+      }
+      parent = parent.getParent();
+    }
+    return null;
+  }
+
+  protected boolean hasChildScreenWithConfig(Screen screen) {
+    if (screen == null) {
+      return false;
+    }
+    for (ScreenContainer sc : screen.getFragment().getChildScreenContainers()) {
+      if (sc instanceof ScreenStack) {
+        // we check only the top screen of stack for header config
+        Screen topScreen = ((ScreenStack) sc).getTopScreen();
```

Comment:
We may already be doing this in othre places but it struct me that we refer to subclasses (stack) from places that operate on a lower level of abstraction (screen) . Maybe we can add some interface method to screen container such that we can avoid testing against the type here.

Reply:
I think we don't do it in other places. I can add `getTopScreen` method to `ScreenContainer` that will return the `Screen` with `ActivityState.ON_TOP`.